### DISCUSSION TITLE
On linux, opening Sleek always starts a new process instead of bringing focus to an already running instance.

### DISCUSSION THREAD

**Discussion by MostHated** [ransome1/sleek#301] [Q&A] [answered]
Hey all, 
    I was trying to setup a script for Autokey to display and hide Sleep via hotkey when I noticed that calling Sleek from my launcher keeps opening up new processes and displaying multiple windows, and multiple taskbar icons.. Most applications in which you have set to run in the background / on the taskbar, you would expect that it would have a single instance and attempting to call the application again would bring focus to an already running instance instead of starting a new process. 

Is there something I can / need to do on my end to produce this type of behavior? If calling the application to try and view it starts a new one anyways, there is really no reason for it to run in the background on the taskbar in the first place.

![](https://i.imgur.com/ItsVRpb.png)

Thanks,
-MH

**Comment by ransome1** [comment, marked as answer]
@MostHated I think this is going to be fixed after this PR is implemented: https://github.com/ransome1/sleek/pull/258

I just wasn't able to review it yet ;)

**Comment by MostHated** [comment]
Oh awesome, glad to hear it is already on the radar. :+1: Being able to quickly open, write, and then get back to what I was doing is one of the most important things to me, because I always forget things quicker than I can think of them, lol.

**Comment by ransome1** [comment]
@MostHated I quickly merged the PR and gave it a quick review. But unfortunately I don't have access to a Linux or Windows machine at the moment. Only MacOS and here this issue doesn't occur.

Could you take a look at https://github.com/ransome1/sleek/releases/tag/v1.1.4-rc.4 and tell me if this solves the issue on your side?

**Reply by MostHated** [reply]
Yeah, sure thing, will do so right now.

**Reply by MostHated** [reply]
For all intents and purposes, it looks like it does, with one caveat. If you attempt to open sleek and its already running and the window is simply behind other windows, it does indeed bring it to the foreground as I had hoped, but if you X out of the window and it is only running on the taskbar and you call sleek again, it seems nothing happens, you have to go to the taskbar item and manually open the window. 

While this behavior is certainly fine with me, I have no issue keeping the window open and simply behind other windows since I can just call it with a hotkey, I am not sure if that is the full and intended behavior, so I just wanted to make sure and note that.

I definitely appreciate the quick resolution to this! Probably the fastest response I have seen. :+1: 
Thanks,
-MH
